modules/adglareBidAdapter.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+import {registerBidder} from '../src/adapters/bidderFactory';
+
+const BIDDER_CODE = 'adglare';
+
+export const spec = {
+  code: BIDDER_CODE,
+  aliases: ['adglare'],
+  isBidRequestValid: function (bid) {
+    return !!(bid.params.domain && bid.params.zID && bid.params.type);
+  },
+  buildRequests: function (validBidRequests, bidderRequest) {
+    let i;
+    let j;
+    let bidRequest;
+    let zID;
+    let domain;
+    let keywords;
+    let url;
+    let type;
+    let availscreen = window.innerWidth + 'x' + window.innerHeight;
+    let screen = ((window.devicePixelRatio || 1) * window.screen.width) + 'x' + ((window.devicePixelRatio || 1) * window.screen.height);
+    let sizes = [];
+    let serverRequests = [];
+    let timeout = bidderRequest.timeout || 0;
+    let referer = bidderRequest.refererInfo.referer || '';
+    let reachedtop = bidderRequest.refererInfo.reachedTop || '';
+    for (i = 0; i < validBidRequests.length; i++) {
+      bidRequest = validBidRequests[i];
+      zID = bidRequest.params.zID;
+      domain = bidRequest.params.domain;
+      keywords = bidRequest.params.keywords || '';
+      type = bidRequest.params.type;
+
+      // Build ad unit sizes
+      if (bidRequest.mediaTypes && bidRequest.mediaTypes[type]) {
+        for (j in bidRequest.mediaTypes[type].sizes) {
+          sizes.push(bidRequest.mediaTypes[type].sizes[j].join('x'));
+        }
+      } else {
+        for (j in bidRequest.sizes) {
+          sizes.push(bidRequest.sizes[j].join('x'));
+        }
+      }
+
+      // Build URL
+      url = 'https://' + domain + '/?' + encodeURIComponent(zID) + '&pbjs&pbjs_version=1';
```

Comment:
Why is the `domain` a bidder param? Will this be different for each bid?

Whatever the domain is, all responses need to be gzipped (See [docs](http://prebid.org/dev-docs/bidder-adaptor.html#bidder-adaptor-Required-Adapter-Conventions)). The responses from the endpoint in your README, try.engine.adglare.net, are not

Reply:
Each of our clients have their own unique subdomain on adglare.net, making it nesserary to add it as a bidder param. I'm checking in isBidRequestValid() if the domain is indeed ours. Yes, try.engine.adglare.net returned a response with Brotli compression. This has been changed now to Gzip for all Prebid responses.

Reply:
Where is this check being done in `isBidRequestValid`?

The only checks I see are these:
https://github.com/prebid/Prebid.js/blob/1108cd1b2e9c254a6ffcdfc2eaff36b0c378d515/modules/adglareBidAdapter.js#L11

Reply:
@msm0504 Indeed I've removed that check since clients were asking if they can use their white label domain for Prebid.js auctions. Since those domains are pointing to our servers, the response will be gzipped. Is this OK for you? I can't validate these white label domains in isBidRequestValid(), since they could be anything. I did however update the function to check whether the domain is a non-empty string.